### PR TITLE
Question filtering by tags

### PR DESCRIPTION
**Pull request by Xupi**

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -62,6 +63,7 @@ import { ProfileViewComponent } from './components/profile-view/profile-view.com
   ],
   providers: [
     DefaultAuthGuard,
+    QuestionsService,
```
Comment (agasior093) [inline src/app/app.module.ts]: I think thats redundant, QuestionsService is @Injectable

File: src/app/components/question-view/question-view.component.ts [hunk 2]
```diff
@@ -1,23 +1,38 @@
...
-  constructor(private questionService: QuestionsService) { }
+  @Input() tagsToFilterBy: Set<string>;
+
+  constructor(private questionService: QuestionsService) {}
 
   ngOnInit() {
-    this.questionService.getAllQuestions().subscribe(payload => this.questions = payload);
+    this.questionService
+      .getAllQuestions()
+      .subscribe(payload => (this.questions = payload));
```
Comment (agasior093) [inline src/app/components/question-view/question-view.component.ts]: What about error handling?

File: src/app/components/question-view/question-view.component.ts [hunk 3]
```diff
@@ -1,23 +1,38 @@
...
   }
 
   toggleAnswer(question: Question, param: boolean) {
     question.revealAnswer = param;
   }
+
+  filterByTags() {
+    this.questionService
+      .getQuestionsFilteredByTags(this.tagsToFilterBy)
+      .subscribe(payload => (this.questions = payload));
```
Comment (agasior093) [inline src/app/components/question-view/question-view.component.ts]: What about error handling?

File: src/app/services/questions.service.ts [hunk 4]
```diff
@@ -20,4 +22,10 @@ export class QuestionsService {
   public getAllQuestions() {
     return this.http.get<any>(EnvironmentConfig.basePath + this.QUESTION_URL, httpOptions());
   }
+
+  public getQuestionsFilteredByTags(tags: Set<string>) {
+    if (!tags || tags.size === 0) { return this.getAllQuestions(); }
+    return this.http.post<any>(EnvironmentConfig.basePath + this.QUESTION_URL +
```
Comment (agasior093) [inline src/app/services/questions.service.ts]: Return concrete type inside <>

**Comment by Xupi** [comment]
About 405 error, will update backend security config for that.

**Pull request merged**